### PR TITLE
Refactor schema generation

### PR DESCRIPTION
**Pull request by ArmanMazdaee**
**What this PR does / why we need it**:
- Refactors schema package to use struct tags
- Adds jsonschema package for generating json schema from struct tags

**Which issue this PR fixes**:
fixes #58

### CODE REVIEW

File: schema.json [hunk 1]
```diff
@@ -1 +1 @@
...
...nies primary activity]","type":"string"},"endDate":{"description":"e.g. 2018-12-29 - [resume.json uses the ISO 8601 date standard]","format":"date","type":"string"},"highlights":{"additionalItems":false,"description":"Specify multiple accomplishments","items":{"description":"e.g. Worked with mobile team at Twitter to develop remote debugging tools for mobile browsers","type":"string"},"type":"array"},"location":{"additionalProperties":false,"description":"e.g. Germany - [Location for this activity]","format":"location","properties":{"lat":{"type":"number"},"long":{"type":"number"}},"type":"object"},"name":{"description":"e.g. XYZ Inc. - [Company name]","type":"string"},"position":{"description":"e.g. Software Engineer - [Position at the company]","type":"string"},"startDate":{"description":"e.g. 2017-06-28 - [resume.json uses the ISO 8601 date standard]","format":"date","type":"string"},"summary":{"description":"Give an overview of your responsibilities at the company","type":"string"},"url":{"description":"e.g. http://xyz.example.com - [Related link to the company website]","format":"uri","type":"string"}},"type":"object"},"type":"array"}},"title":"Resumic Schema","type":"object"}
```
Comment (stp-ip) [inline schema.json]: In the long run we could think about solutions to make it more human readable as a json file. :smile:

File: schema/schema.go [hunk 2]
```diff
@@ -6,161 +6,161 @@ type locationSchema struct {
...
+	StartDate   string         `jsonschema:"startDate;description:e.g. 2017-06-29 - [resume.json uses the ISO 8601 date standard];format:date"`
+	EndDate     string         `jsonschema:"endDate;description:e.g. 2013-06-29 - [resume.json uses the ISO 8601 date standard];format:date"`
 	Score       struct {
-		Scoretype  string `jsonschema:"scoretype,description:eg. GPA or PERCENTAGE - [Add score type]"`
-		Scorevalue string `jsonschema:"scorevalue,description:eg. 3.4/4.0 - [Add obtained score/total score]"`
-	} `jsonschema:"score,additionalProperties"`
-	Courses []string `jsonschema:"courses,description:List notable courses/subjects,items_description:e.g. CS302 - Introduction to Algorithms - [Add course name]"`
-	Honors  []string `jsonschema:"honors,description:List education honours,items_description:e.g. Magna Cum Laude"`
+		Scoretype  string `jsonschema:"scoretype;description:eg. GPA or PERCENTAGE - [Add score type]"`
+		Scorevalue string `jsonschema:"scorevalue;description:eg. 3.4/4.0 - [Add obtained score/total score]"`
```
Comment (stp-ip) [inline schema/schema.go]: Why not use `ScoreType`, `ScoreValue`, `scoreType` and `scoreValue`?

Comment (stp-ip) [reply schema/schema.go]: Might actually be shortened to `Type` and `Value` as it's within the `Score` struct. Thoughts?

File: schema/schema.go [hunk 3]
```diff
@@ -6,161 +6,161 @@ type locationSchema struct {
...
+	Company   string `jsonschema:"company;description:e.g. Xyz"`
+	Position  string `jsonschema:"position;description:e.g. Senior Software Engineer"`
+	Reference string `jsonschema:"reference;description:e.g. Joe blogs was a great employee, who turned up to work at least once a week. He exceeded my expectations when it came to doing nothing."`
 }
 
 type languageSchema struct {
-	Language string  `jsonschema:"language,description:e.g. English - [Name of language]"`
-	Score    float64 `jsonschema:"score,description:e.g. 20 - [Score for the language]"`
+	Language string  `jsonschema:"language;description:e.g. English - [Name of language]"`
+	Score    float64 `jsonschema:"score;description:e.g. 20 - [Score for the language]"`
```
Comment (stp-ip) [inline schema/schema.go]: Maybe reuse the `Score` struct? That way one could add the type of a score.

File: schema.json (hunk 1 above)
Comment (stp-ip) [reply schema.json]: Awesome. I was too slow with the review :)

File: schema/schema.go (hunk 2 above)
Comment (ArmanMazdaee) [reply schema/schema.go]: Fixed in bd74350

File: schema/schema.go (hunk 3 above)
Comment (ArmanMazdaee) [reply schema/schema.go]: Fixed in 39d8490

**Comment by stp-ip** [comment]
@erbesharat @alarsyo happy to get another review or general thoughts on this. For me it's lgtm and iterate.

**Comment by alarsyo** [comment]
Reviewing tonight.

**Comment by stp-ip** [comment]
\o/

**Pull request merged**